src/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp

```diff
@@ -471,6 +471,94 @@ bool r1cs_ppzksnark_affine_verifier_weak_IC(const r1cs_ppzksnark_verification_ke
                                             const r1cs_ppzksnark_primary_input<ppT> &primary_input,
                                             const r1cs_ppzksnark_proof<ppT> &proof);
 
+/****Batch and probabilistic verification 
+ * using randomness and the bilinearity of the pairing operation, the Pinocchio verifier can be made more efficient
+ * with the price of making the verification procedure probabilistic, introducing a negligible chance of accepting a bad proof.
+ * In a similar way verifying a batch of proofs can be bundled into verifying a single pairing equation using randomness
+ * such that the verifier always accepts if all proofs in the batch are valid, and the verifier accepts with a negligible 
+ * probability when one of the proofs in the batch is invalid.
+ * The methods below implemement these verifiers,
+ * and the logic behind them are described in  https://github.com/arielgabizon/libsnark/blob/multimiller/src/zk_proof_systems/batchverification.pdf
```

Comment:
Change this to ``src/zk_proof_systems/batchverification.pdf`` rather than giving the full URL to your repo. Similarly on lines 490 and 513.